### PR TITLE
[RFC] Hello Vite

### PR DESCRIPTION
**Pull request by ohbarye**
Hello @mobu-of-the-world/maintainers , this pull request aims to migrate this project's build tool from react-scripts (= create-react-app including Webpack) to [Vite](https://vitejs.dev/).

### What's Vite?

First, Vite is a build tool for frontend JavaScript applications. You must know Webpack, Parcel, Rollup etc. Vite follows the genealogy.

It uses [esbuild](https://esbuild.github.io/) for a development web server so that it makes quite faster server bootstrap and HMR. Plus, it uses [Rollup](https://rollupjs.org/) for a production build.

### Why Vite?

Webpack and create-react-app are not bad tools even in 2022, I think. But they're just _not bad_, and not great. In 2022, there's happening another paradigm for JS build. That's the Native ESM based build. See [Why Vite?](https://vitejs.dev/guide/why.html) in the public document.

The following depicts the core difference between 2 paradigms. Our good old webpack, rollup, parcel are the upper one, and Vite (technically, esbulid) and snowpack belongs to the lower.

image

### Vite's momentum

Vite is the most satisfying build tool in the State of JS 2021 survey. (I feel it's a bit strange to treat Vite / esbuild / Rollup in the same line since Vite includes esbuild and Rollup, though.)

https://2021.stateofjs.com/en-US/libraries/build-tools

image

### Why this project should use Vite?

...

To be honest, there's no strong pressure to adopt Vite in this project since Vite brings out its power in a relatively bigger project than this one.

But I assume mobu maitainers have a kind of challenging spirit to try new things. So I brought the very new thing that may interest you. I'm looking forward to hearing your opinion on this.

## Scope

Just improving the developer experience. No influence on users.

## Changes

I mostly followed the procedure written in https://darekkay.com/blog/create-react-app-to-vite/.

### CODE REVIEW

**Comment by pankona** [comment]
@ohbarye 

Thank you for your great suggestion! Please let me check!

**Comment by pankona** [comment]
It works...! Brazing fast to start dev server...!

File: index.html [hunk 1]
```diff
@@ -2,36 +2,24 @@
...
+    <link rel="icon" href="/favicon.ico" />
     <meta
       name="viewport"
       content="width=device-width,initial-scale=1.0,minimum-scale=1.0"
     />
     <meta name="theme-color" content="#000000" />
     <meta
       name="description"
-      content="Web site created using create-react-app"
+      content="A web application to manage participants and timer for mob-programming!"
```
Comment (pankona) [inline index.html:13]: Great improvement, thanks!!

**Review by pankona — APPROVED** [review]

**Comment by pankona** [comment]
I don't see any obstacle to use Vite for mobu. I love it!
@kachick, hey, how do you think?

**Comment by pankona** [comment]
@ohbarye 

I resolved conflict that is caused by the yarn.lock, by merging main branch. Thanks.

**Comment by pankona** [comment]
Let's merge it. Let's give it a try!

**Comment by ohbarye** [comment]
Thank you @pankona san!
I hope this change could brow the mobu team's mind. 💨

**Comment by kachick** [comment]
Thank you for integrated the good tool!

Few month ago, I and @pankona ojisan had tackled to integrate Vite into another 🍝 webpack project and both givem upped. `Your power > pankona + kachick` 🤣

**Comment by ohbarye** [comment]
Don't flatter me! 😄 
I hope also [my recent article](https://blog.smartbank.co.jp/entry/2022/04/25/create-react-app-to-vite) would help your project's success!

**Pull request merged**